src/components/scenes/WaitingRoom.js

```diff
@@ -0,0 +1,228 @@
+import { useContext, useEffect } from "react";
+import { Text, TouchableOpacity, View, ScrollView } from "react-native";
+import { transact, tx, useQuery, id } from "@instantdb/react-native";
+import Toast from "react-native-root-toast";
+import * as Clipboard from "expo-clipboard";
+
+import { GAME_IN_PROGRESS, generateGameColors } from "@/game";
+import SafeView from "@/components/shared/SafeView";
+import { avatarColor } from "@/utils/profile";
+import { now } from "@/utils/time";
+import {
+  primaryBackgroundColor as bgColor,
+  RegularButton,
+} from "@/components/shared/styles";
+import {
+  LoadingPlaceholder,
+  ErrorPlaceholder,
+} from "@/components/shared/Placeholder";
+import { UserContext } from "@/Context";
+
+function userSort(a, b) {
+  // Host comes first
+  if (a.isHost) return -1;
+  if (b.isHost) return 1;
+
+  // You comes second
+  if (a.isYou) return -1;
+  if (b.isYou) return 1;
+
+  // Sort the rest alphabetically by handle
+  return a.handle.localeCompare(b.handle);
+}
+
+function startMultiplayerGame(room) {
+  const gameId = id();
+  const { users } = room;
+  const colors = generateGameColors();
+
+  const playerIds = users
+    .filter((u) => u.id === room.hostId || room.readyIds.includes(u.id))
+    .map((u) => u.id);
+  const createGame = tx.games[gameId].update({
+    status: GAME_IN_PROGRESS,
+    playerIds,
+    colors,
+    created_at: now(),
+  });
+  const addUserGameLinks = users.map((u) =>
+    tx.games[gameId].link({ users: u.id })
+  );
+  const createPoints = playerIds.map((playerId) =>
+    tx.points[id()].update({ val: 0, userId: playerId }).link({ games: gameId })
+  );
+  const updateRoom = tx.rooms[room.id]
+    .update({
+      currentGameId: gameId,
+      readyIds: [],
+    })
+    .link({ games: gameId });
+
+  transact([createGame, ...addUserGameLinks, ...createPoints, updateRoom]);
+}
+
+function InviteButton({ code }) {
+  async function copy(code) {
+    await Clipboard.setStringAsync(
+      `Let's play Stroopwafel! My game code is: ${code}`
+    );
+    Toast.show("Game code copied! Send it over to your friends :)", {
+      duration: Toast.durations.LONG,
+    });
+  }
+  return (
+    <RegularButton onPress={() => copy(code)}>Invite Friends</RegularButton>
+  );
+}
+
+function UserPill({ user, room, isReady, isAdmin }) {
+  const { isYou, isHost, id: userId, handle } = user;
+  const { id: roomId, readyIds, kickedIds } = room;
+
+  // Avatar
+  const avatarStyle = avatarColor(handle);
+
+  // Title
+  let title = [];
```

Comment:
nit: maybe this can be a fn

Reply:
@stopachka Curious, what's the benefit of putting this in an `fn`?

Reply:
May be too much of an it, but my thinking was: 

The intent for L86-L93 is to create a string. But, we use mutation + array. Maybe this detail is better abstracted away somewhere else

Reply:
Noted!